### DISCUSSION TITLE
Request for Sample Helm Chart and Auto-Scaling Clarification

### DISCUSSION THREAD

**Discussion by mohamedsaleem18** [nats-io/nats-server#6080] [Q&A]
We are currently using a 3-node NATS cluster on Red Hat OpenShift (on-premises). The cluster consists of 3 NATS server pods, with a headless service exposed to client applications deployed within the OpenShift cluster. For client applications deployed outside of the OpenShift cluster, the 3 seed URLs should be exposed using ingress or routes.

Could you please provide a sample Helm chart for this setup?

Additionally, in the case of auto-scaling, will new ports be automatically assigned?

**Comment by Jarema** [comment]
We are maintaining official Helm charts.

Please check here: https://github.com/nats-io/k8s/tree/main/helm/charts/nats

I would reconsider autoscaling - if you are using Core NATS, there is a really high chance you will  not need it, as NATS is really performant, and if you're using JetStream - autoscaling would simply not work, as you need to consider Raft assets before  removing removing nodes (scaling down).

**Comment by mohamedsaleem18** [comment]
Thank you for your response. Referring nats.io Helm chart, a 3-node NATS cluster is configured with a single NodePort (load balancer) for client applications outside of OpenShift cluster to communicate. Applications deployed on OpenShift cluster uses headless service (ClusterIP). Please refer to the Helm chart configuration below for the node port. NATS.io documentation recommends that client applications use seed URLs for each NATS server individually. Could you please provide an example Helm chart that configures 3 separate NodePorts, one for each NATS server?

```
nats:
  port: 4222

service:
  enabled: true
  merge:
    spec:
      type: NodePort
  ports:
    nats:
      enabled: true
      nodePort: 30518
```
